oauth/opts.go

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright (c) 2020 SSH Communications Security Inc.
+//
+// All rights reserved.
+//
+
+package oauth
+
+import (
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+
+	"github.com/BurntSushi/toml"
+)
+
+// Option is configuration applied to the client
+type Option func(*tAuth) *tAuth
+
+// Access setups client access key
+func Access(access string) Option {
+	return func(auth *tAuth) *tAuth {
+		if access != "" {
+			auth.access = access
+		}
+		return auth
+	}
+}
+
+// Secret setups clients secret key
+func Secret(secret string) Option {
+	return func(auth *tAuth) *tAuth {
+		if secret != "" {
+			auth.secret = secret
+		}
+		return auth
+	}
+}
+
+// Digest setups client secret digest
+func Digest(oauthAccess, oauthSecret string) Option {
+	return func(auth *tAuth) *tAuth {
+		if oauthAccess != "" && oauthSecret != "" {
+			auth.digest = base64.StdEncoding.EncodeToString([]byte(oauthAccess + ":" + oauthSecret))
+		}
+		return auth
+	}
+}
+
+// UseConfigFile setup credential from tol file
+func UseConfigFile(path *string) Option {
+	return func(auth *tAuth) *tAuth {
+		type config struct {
+			AuthClientID     string `toml:"oauth_client_id"`
+			AuthClientSecret string `toml:"oauth_client_secret"`
+			ClientID         string `toml:"api_client_id"`
+			ClientSecret     string `toml:"api_client_secret"`
+		}
+		var file struct {
+			Auth config
+		}
+
+		if path == nil {
+			return auth
+		}
+
+		f, err := os.Open(*path)
+		if err != nil {
+			panic(err)
+		}
+		defer f.Close()
+
+		data, err := ioutil.ReadAll(f)
+		if err != nil {
+			panic(err)
+		}
+
+		err = toml.Unmarshal(data, &file)
+		if err != nil {
```

Comment:
non-existing file is crashes the application, but malformed content is silently accepted?

Reply:
Silently, ignored. But let is crash!